### PR TITLE
[dsub] Implement changes to returned job models for capabilities update

### PR DESCRIPTION
**Pull request by bfcrampton**

### CODE REVIEW

File: servers/dsub/jobs/controllers/jobs_controller.py [hunk 1]
```diff
@@ -245,17 +233,33 @@ def _handle_http_error(error, proj_id):
...
 
 
 def _provider_type():
     return current_app.config['PROVIDER_TYPE']
+
+
+def _query_jobs_result(job, project_id=None):
+    return QueryJobsResult(
+        id=job_ids.dsub_to_api(project_id, job['job-id'], job.get(
+            'task-id', '')),
```
Comment (calbach) [inline servers/dsub/jobs/controllers/jobs_controller.py]: nit: I'd probably make it accept `None` for task-id

File: servers/dsub/jobs/controllers/utils/extensions.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+        Args:
+            job: A dict with dsub job metadata
+
+        Returns:
+            ExtendedFields: Populated extensions on the job
+    """
+    return ExtendedFields(
+        user_id=job['user-id'],
+        status_detail=job['status-detail'],
```
Comment (calbach) [inline servers/dsub/jobs/controllers/utils/extensions.py]: Are `status-detail` and `last-update` sure things? Will they ever be `None`? Maybe safer to just use `get()` instead

File: servers/dsub/jobs/controllers/utils/labels.py [hunk 3]
```diff
@@ -10,15 +10,8 @@ def dsub_to_api(job):
     # Put any dsub specific information into the labels. These fields are
```
Comment (calbach) [inline servers/dsub/jobs/controllers/utils/labels.py]: nit: drop comment

File: servers/dsub/jobs/controllers/utils/labels.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-def dsub_to_api(job):
+def get_labels(job):
```
Comment (calbach) [inline servers/dsub/jobs/controllers/utils/labels.py]: nit: I like the previous naming better I think, it's basically just stripping out things that are not labels in the API.

File: servers/dsub/jobs/controllers/jobs_controller.py (hunk 1 above)
Comment (bfcrampton) [reply servers/dsub/jobs/controllers/jobs_controller.py]: Yup, already does not sure why it was like this.

File: servers/dsub/jobs/controllers/utils/extensions.py (hunk 2 above)
Comment (bfcrampton) [reply servers/dsub/jobs/controllers/utils/extensions.py]: Good point, changed to `get()`

File: servers/dsub/jobs/controllers/utils/labels.py (hunk 4 above)
Comment (bfcrampton) [reply servers/dsub/jobs/controllers/utils/labels.py]: Done.

File: servers/dsub/jobs/controllers/utils/labels.py (hunk 3 above)
Comment (bfcrampton) [reply servers/dsub/jobs/controllers/utils/labels.py]: Done.

**Pull request merged**